sal-service/src/main/java/org/ow2/proactive/sal/service/service/PersistenceService.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package org.ow2.proactive.sal.service.service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.ow2.proactive.sal.model.*;
+import org.ow2.proactive.scheduler.common.exception.NotConnectedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.extern.log4j.Log4j2;
+
+
+@Log4j2
+@Service("PersistenceService")
+public class PersistenceService {
+    @Autowired
+    private PAGatewayService paGatewayService;
+
+    @Autowired
+    private CloudService cloudService;
+
+    @Autowired
+    private RepositoryService repositoryService;
+
+    @Autowired
+    private ClusterService clusterService;
+
+    @Autowired
+    private EdgeService edgeService;
+
+    /**
+     * Clean all clusters, clouds, edge devices, and database entries.
+     * @param sessionId A valid session id
+     */
+    public boolean cleanAll(String sessionId) throws NotConnectedException {
```

Comment:
In my opinion this method should be in a transaction and if one of the delete fails, it should rollback.
This operation should be atomic as here we can have a failure response but in reality we have deleted parts of what we want.

Reply:
yes, i was considering to do it. thanks for a reminder ;)

Reply:
in the end it is not possible to use it, as it brakes async processes, and does not roll back the changes on the side of iaas connector. i will keep it as it was and see if there is something to do about it in later imporvements of sal ;)